domain/storageprovisioning/state/volume.go

```diff
@@ -0,0 +1,361 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"database/sql"
+	"maps"
+
+	"github.com/canonical/sqlair"
+
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/core/watcher/eventsource"
+	"github.com/juju/juju/domain"
+	"github.com/juju/juju/domain/life"
+	"github.com/juju/juju/domain/storageprovisioning"
+	"github.com/juju/juju/internal/errors"
+)
+
+// GetVolumeAttachmentIDs returns the [storageprovisioning.VolumeAttachmentID]
+// information for each volume attachment uuid supplied. If a uuid does not
+// exist or isn't attached to either a machine or a unit then it will not exist
+// in the result.
+func (st *State) GetVolumeAttachmentIDs(
+	ctx context.Context, uuids []string,
+) (map[string]storageprovisioning.VolumeAttachmentID, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
```

Comment:
Return early if the uuids slice is empty.

Reply:
Done & Done